src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,445 @@
+    function calcPercentage(comp) {
+      return (
+        (comp.currentValue - comp.minValue) *
+        100 /
+        (comp.maxValue - comp.minValue)
+      );
+    }
+    const childRanges = Array.from(
+      this.element.querySelectorAll("gx-gauge-range")
+    );
+    this.maxValue = this.totValues + this.minValue;
+    this.totValues = this.maxValue - this.minValue;
+    if (this.gaugeType === "Circle") {
+      const svgRanges = [];
+      let acumul = 0;
+      //////////////////////////////////////////
+      function calcPositionRange(preValue, value) {
+        acumul += preValue;
+        return value + acumul;
+      }
+      //////////////////////////////////////////
+      for (let i = childRanges.length - 1; i >= 0; i--) {
+        svgRanges.push(
+          <circle
+            r="39.59%"
+            cx="50%"
+            cy="50%"
+            stroke={childRanges[i].color}
+            stroke-dasharray={
+              "" +
+              (2.488 /
+                (this.totValues / 100) *
+                calcPositionRange(
+                  !!childRanges[i + 1]
+                    ? parseInt(childRanges[i + 1].getAttribute("amount"), 10)
+                    : 0,
+                  parseInt(childRanges[i].getAttribute("amount"), 10)
+                ) +
+                ", 248.16")
+            }
+            fill="none"
+            amount={childRanges[i].amount}
+            stroke-width={"" + this.calcThickness() + "%"}
+          />
+        );
+      }
+      svgRanges.reverse();
+      ////////////////////////////////////////
+      return (
+        <div
+          class="svgContainer"
+          style={{
+            height: this.minorSize * 1 + "px",
+            width: this.minorSize * 1 + "px"
+          }}
+        >
+          <svg width="100%" height="100%" viewBox="0 0 100 100">
+            {svgRanges.map(gaugeRange => gaugeRange)}
+          </svg>
+          <div
+            class="gaugeContainer"
+            style={{
+              border: this.styleBorder
+                ? this.styleBorderWidth + "px solid " + this.styleBorderColor
+                : "",
+              "box-shadow":
+                (this.minorSize <= 300 && this.thickness <= 25) ||
+                !this.styleShadow
+                  ? "none"
+                  : "",
+              height:
+                this.minorSize * 0.8 +
+                (this.calcThickness() * (this.minorSize / 100) -
+                  this.minorSize / 100) +
+                "px",
+              width:
+                this.minorSize * 0.8 +
+                (this.calcThickness() * (this.minorSize / 100) -
+                  this.minorSize / 100) +
+                "px"
+            }}
+          />
+          {this.showValue ? (
+            <span
+              class="marker"
+              style={{
+                display: this.showValue ? "" : "none",
+                height:
+                  this.minorSize * 0.795 -
```

Comment:
Please consider using [string literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) in cases like this.